lib/AST/Decl.cpp

```diff
@@ -407,6 +408,19 @@ DeclAttributes Decl::getSemanticAttrs() const {
   return getAttrs();
 }
 
+void Decl::attachParsedAttrs(DeclAttributes attrs) {
+  ASSERT(getAttrs().isEmpty() && "attaching when there are already attrs?");
+
+  for (auto *attr : attrs.getAttributes<DifferentiableAttr>())
+    attr->setOriginalDeclaration(this);
+  for (auto *attr : attrs.getAttributes<DerivativeAttr>())
+    attr->setOriginalDeclaration(this);
+  for (auto attr : attrs.getAttributes<ABIAttr, /*AllowInvalid=*/true>())
+    attr->createInverse(this);
```

Comment:
Nice! Thank you for doing the autodiff things too!

Reply:
Keep in mind that I haven't actually implemented these attributes in ASTGen, but you should be able to drop in an implementation for them now. :D